src/core/lib/transport/transport.h

```diff
@@ -507,6 +517,21 @@ class Transport : public InternallyRefCounted<Transport> {
   struct RawPointerChannelArgTag {};
   static absl::string_view ChannelArgName() { return GRPC_ARG_TRANSPORT; }
 
+  // Though internally ref counted transports expose their "Ref" method to
```

Comment:
It looks like this is necessary for the connectivity watcher created by the server code.  But in this case, we're really not interested in a general-purpose connectivity state watcher; all we want to know is when the connection fails.  Could we instead just pass a callback to the transport that it will invoke when the transport is closed and should be dropped from the server?

Reply:
"just pass a callback" introduces significantly more complexity than is here... I'd like to get away from the place that there's a complicated dance that each transport needs to do to coordinate its shutdown with other systems.

Reply:
Good point.  Okay, this is fine for now, but I think that as part of @yashykt's refactoring of the transport server code, I'm going to push for structuring this in a way that doesn't require exposing these methods.  And I think that will actually fall out of that work naturally, since I think we will wind up needing to store some per-connection state that is not held in the transport itself.